### PR TITLE
Adds support for Ion 1.1 delimited containers and inline symbols.

### PR DESCRIPTION
**Pull request by tgregg**
### Description of changes:

Example output for command `ion-java-benchmark-cli write --warmups 2 --iterations 2 --forks 2 --mode AverageTime --ion-minor-version 0 --ion-minor-version 1 --ion-inline-symbols all --ion-delimited-containers none --ion-inline-symbols none --ion-delimited-containers all --ion-flush-period 500  log.ion`

Note: N = `--ion-inline-symbols`, C = `--ion-delimited-containers`

```
Benchmark                                   (input)                                                         (options)  Mode  Cnt          Score          Error   Units

Bench.run                                   log.ion    write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:ALL,C:ALL}  avgt    4        254.567 ±       15.968   ms/op
Bench.run:Heap usage                        log.ion    write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:ALL,C:ALL}  avgt    4        446.666 ±      250.743      MB
Bench.run:Serialized size                   log.ion    write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:ALL,C:ALL}  avgt    4        116.496                     MB

Bench.run                                   log.ion                write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:0}  avgt    4        202.997 ±       67.386   ms/op
Bench.run:Heap usage                        log.ion                write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:0}  avgt    4        367.582 ±      374.341      MB
Bench.run:Serialized size                   log.ion                write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:0}  avgt    4         21.396                     MB

Bench.run                                   log.ion   write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:NONE,C:ALL}  avgt    4        174.974 ±       13.372   ms/op
Bench.run:Heap usage                        log.ion   write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:NONE,C:ALL}  avgt    4        342.455 ±      426.056      MB
Bench.run:Serialized size                   log.ion   write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:NONE,C:ALL}  avgt    4         22.304                     MB

Bench.run                                   log.ion   write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:ALL,C:NONE}  avgt    4        308.391 ±       61.220   ms/op
Bench.run:Heap usage                        log.ion   write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:ALL,C:NONE}  avgt    4        418.646 ±     1104.092      MB
Bench.run:Serialized size                   log.ion   write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:ALL,C:NONE}  avgt    4        116.526                     MB

Bench.run                                   log.ion  write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:NONE,C:NONE}  avgt    4        188.626 ±       19.387   ms/op
Bench.run:Heap usage                        log.ion  write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:NONE,C:NONE}  avgt    4        341.320 ±      334.202      MB
Bench.run:Serialized size                   log.ion  write::{f:ION_BINARY,t:FILE,d:500,a:STREAMING,V:1,N:NONE,C:NONE}  avgt    4         19.311                     MB
```

----
_**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**_

### CODE REVIEW

File: src/com/amazon/ion/benchmark/Gradient.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.amazon.ion.benchmark;
+
+public enum Gradient {
```
Comment (popematt) [inline src/com/amazon/ion/benchmark/Gradient.java:6]: suggestion— "Gradient" is not a self-explanatory name. (It's a reasonably well-defined term, but it's lacking context here.) Can you add some javadoc?

**Review by popematt — APPROVED** [review]
Just one issue. I'm fine with it if you want to address it in a followup PR.

File: src/com/amazon/ion/benchmark/Gradient.java (hunk 1 above)
Comment (tgregg) [reply src/com/amazon/ion/benchmark/Gradient.java:6]: Sure, but I'm also not set on this name at all. I just want a name for the words that will be used to describe a range from none to all (and maybe values in-between in the future). Any ideas?

Comment (popematt) [reply src/com/amazon/ion/benchmark/Gradient.java:6]: So, if you really want it to just represent a number from `0.0` to `1.0`, the "gradient" or just about any synonym is fine. However, if we want to be able to e.g. test with other strategies, such as "intern field names and annotations, but inline symbol values" or "only use length prefixes for top-level containers", then I don't think that "gradient" would make sense.

Comment (tgregg) [reply src/com/amazon/ion/benchmark/Gradient.java:6]: I decided to keep "Gradient" now, for lack of something better. I added a short description. If we end up needing more specific options related to inline symbols or delimited containers, we'll likely need to swap this out for a more targeted enum anyway.

**Pull request merged**